anta/runner.py

```diff
@@ -49,62 +44,81 @@ def log_cache_statistics(devices: list[AntaDevice]) -> None:
-
-    Returns
-    -------
-        any: ResultManager object gets updated with the test results.
-
     """
     if not catalog.tests:
         logger.info("The list of tests is empty, exiting")
         return
     if len(inventory) == 0:
         logger.info("The inventory is empty, exiting")
         return
+
+    # Filter the inventory based on tags and devices parameters
+    inventory = inventory.get_inventory(
+        tags=tags,
+        devices=devices,
+    )
     await inventory.connect_inventory()
-    devices: list[AntaDevice] = list(inventory.get_inventory(established_only=established_only, tags=tags).values())
-
-    if not devices:
-        msg = (
-            f"No device in the established state '{established_only}' {f'matching the tags {tags} ' if tags else ''}was found. "
-            "There is no device to run tests against, exiting"
-        )
-        logger.info(msg)
+
+    # Remove devices that are unreachable
+    inventory = inventory.get_inventory(established_only=established_only)
+
+    if not inventory.devices:
+        msg = f"No reachable device {f'matching the tags {tags} ' if tags else ''}was found."
+        logger.warning(msg)
         return
     coros = []
+
+    # Select the tests from the catalog
+    if tests:
+        catalog = AntaCatalog(catalog.get_tests_by_name(tests))
+
     # Using a set to avoid inserting duplicate tests
-    tests_set: set[AntaTestRunner] = set()
-    for device in devices:
+    selected_tests: set[AntaTestRunner] = set()
+
+    # Create AntaTestRunner tuples from the tags
+    for device in inventory.devices:
         if tags:
             # If there are CLI tags, only execute tests with matching tags
-            tests_set.update((test, device) for test in catalog.get_tests_by_tags(tags))
+            selected_tests.update((test, device) for test in catalog.get_tests_by_tag(tags))
         else:
-            # If there is no CLI tags, execute all tests without filters
-            tests_set.update((t, device) for t in catalog.tests if t.inputs.filters is None or t.inputs.filters.tags is None)
+            # If there is no CLI tags, execute all tests that do not have any filters
+            selected_tests.update((t, device) for t in catalog.tests if t.inputs.filters is None or t.inputs.filters.tags is None)
 
             # Then add the tests with matching tags from device tags
-            tests_set.update((t, device) for t in catalog.get_tests_by_tags(device.tags))
-
-    tests: list[AntaTestRunner] = list(tests_set)
+            selected_tests.update((t, device) for t in catalog.get_tests_by_tag(device.tags))
 
-    if not tests:
-        msg = f"There is no tests{f' matching the tags {tags} ' if tags else ' '}to run on current inventory, exiting"
-        logger.info(msg)
+    if not selected_tests:
+        msg = f"There is no tests{f' matching the tags {tags} ' if tags else ' '}to run in the current test catalog, please verify your inputs."
```

Comment:
probablly need to keep the notion of current inventory as this is potentially a dual filtering
like runing a given test notfor spine on a spine device.